scrapy_poet/page_input_providers.py

```diff
@@ -263,3 +283,81 @@ class ResponseUrlProvider(PageObjectInputProvider):
     def __call__(self, to_provide: Set[Callable], response: Response):
         """Builds a ``ResponseUrl`` instance using a Scrapy ``Response``."""
         return [ResponseUrl(url=response.url)]
+
+
+class ItemProvider(PageObjectInputProvider):
+
+    name = "item"
+
+    def __init__(self, injector):
+        super().__init__(injector)
+        self.registry = self.injector.registry
+
+        # The key that's used here is the ``scrapy.Request`` instance to ensure
+        # that the cached instances under it are properly garbage collected
+        # after processing such request.
+        self._cached_instances = WeakKeyDictionary()
+
+    def provided_classes(self, cls):
+        """If the item is in any of the ``to_return`` in the rules, then it can
+        definitely provide by using the corresponding page object in ``use``.
```

Comment:
```suggestion
        """If the item is in any of the ``to_return`` in the rules, then it can
        be provided by using the corresponding page object in ``use``.
```

Reply:
a46b1e2